setup-scripts/install.sh

```diff
@@ -81,15 +121,68 @@ msg "Installing node.js@8.11" $WHITE
+    LINUX_DISTRO=$(cat /etc/*-release | grep -E '^ID=' | sed 's/^.*=//' | tr -d \")
+    msg "Looks like you're using Linux." $GREEN
+    msg "To install Docker using the convenience script for Linux, run" $GREEN
+    msg "    $ curl -fsSL https://get.docker.com -o get-docker.sh" $WHITE
+    msg "    $ sudo sh get-docker.sh" $WHITE
+    msg "Alternatively, you can follow Docker's installation steps for Linux (https://docs.docker.com/install/linux/docker-ce/$(echo $LINUX_DISTRO)/)" $GREEN
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+    msg "Looks like you're using Mac." $GREEN
+    msg "You can use the following link to download and install Docker for Mac Version 2.0.0.3 or greater (this installs Docker Community Edition and Docker Compose)" $GREEN
+    msg "https://docs.docker.com/docker-for-mac/release-notes/#docker-community-edition-2003-2019-02-15" $WHITE
+  else
+    msg "Please install Docker before continuing." $GREEN
+  fi
+  exit 1
+fi
+
+# Ensure version of Docker meets minimum requirements. Warn if version is not the latest
+DOCKER_VERSION=$(docker version | grep Version | head -n 1 | grep -oE '[.0-9]*$')
+compare_versions $DOCKER_VERSION $MIN_DOCKER_VERSION
+if [[ $? == 2 ]]; then
+  msg "Your version of Docker is older than the minimum required version. Please update to the latest version of Docker." $RED
   exit 1
 fi
 
+compare_versions $DOCKER_VERSION $CURRENT_DOCKER_VERSION
+if [[ $? == 2 ]]; then
+  msg "Your version of Docker isn't up to date. Please update to the latest version." $YELLOW
+fi
+
 if [ "$(command -v docker-compose)" == "" ]; then
-  msg "Docker Compose is not installed. Please install it before continuing" $RED
+  msg "Docker Compose is not installed." $RED
+  if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    msg "To install Docker Compose using Linux, run" $GREEN
+    msg '    $ curl -L "https://github.com/docker/compose/releases/download/1.23.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose' $WHITE
+    msg "    $ sudo chmod +x /usr/local/bin/docker-compose" $WHITE
+    msg "Alternatively, you can follow Docker's installation steps for Linux (https://docs.docker.com/compose/install/)" $GREEN
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+    msg "Looks like you're using Mac." $GREEN
+    msg "You can use the following link to install Docker Compose for Mac." $GREEN
+    msg "https://docs.docker.com/compose/install/" $WHITE
+  else
+    msg "Please install Docker Compose before continuing" $RED
+  fi
   exit 1
 fi
 
+# Ensure version of Docker Compose meets minimum requirements. Warn if version is not the latest
+DOCKER_COMPOSE_VERSION=$(docker-compose version | grep 'docker-compose version' | sed 's/,.*//' | grep -oE '[.0-9]*$')
+compare_versions $DOCKER_COMPOSE_VERSION $MIN_DOCKER_COMPOSE_VERSION
+if [[ $? == 2 ]]; then
+  msg "Your version of Docker Compose is older than the minimum required version. Please update to the latest version of Docker Compose." $RED
```

Comment:
Just like above, include the actual versions in here. And also include a link to docs or something similar to give them a path to upgrade.